src/parse.ts

```diff
@@ -76,12 +76,20 @@ export function parse(messageDefinition: string, options: ParseOptions = {}): Me
       : buildType(definitionLines, ROS1_GRAMMAR),
   );
 
+  // Filter out duplicate types.
+  // This will avoid that searching a type by name will return more than one result.
+  const seenTypes = new Set<string>();
+  const uniqueTypes = types.filter((definition) => {
```

Comment:
why are we deduping instead of making the search function able to return multiple or return the first that matches?
Should having duplicate types be a schema error?

Reply:
+1 - seems weird to me that we would have duplicate types. At least it seems like we should confirm they are identical and throw an error if not (or always throw an error)

Reply:
Turns out that duplicates were erroneously produced by `mcap convert`, so we have to handle that case. I changed it now to a deep comparison instead of only checking the name.